fix invalid image path and jenkins url in the stacktrace

jugend opened the pull request:
- On click and exception screenshots paths shown in the stackstrace is invalid, i.e.

``` js
// Local screenshot path (invalid)
      at Async.drainQueues (/Users/hsiatono/codes/paypal/hsiatono/fixo/node_modules/bluebird/js/main/async.js:15:14)
  nemo-screenshot::{"imageName":"test/functional/report/_prefill__sender__Sender_create_account_happy_flow_should_complete_successfully___US.png","imagePath":"/Users/hsiatono/codes/paypal/consumerweb/conspartneronbnodeweb/test/functional/report/Users/hsiatono/codes/paypal/consumerweb/conspartneronbnodeweb/test/functional/report/_prefill__sender__Sender_create_account_happy_flow_should_complete_successfully___US.png"}::nemo-screenshot

// Jenkin URL generated (invalid)
https://ci.paypalcorp.com/conspartneronbnodeweb-ci-6518/job/develop-full-regression-tests/ws/test/functional/report//jobvolume/conspartneronbnodeweb-ci-6518/workspace/develop-full-regression-tests/test/functional/report/_autofill_Erste___Create_Account_with_multiple__bank__card_should_onboard_successfully_with_1_bank___1_card_for__DE.png
```
- Also added archived image URL for Jenkins, as most of the times we archived the artifacts, sample output:

``` js
Error: [drivex.waitForElement] Element not locatable
    at node_modules/nemo-view/node_modules/selenium-drivex/index.js:54:15
    ...
nemo-screenshot (workspace): https://ci.paypalcorp.com/conspartneronbnodeweb-ci-6518/job/hsiatono-smoke-tests/ws/test/functional/report/_card__add_card__Add_card_variations_should_add_card___maestro_GB.png
nemo-screenshot (archived): https://ci.paypalcorp.com/conspartneronbnodeweb-ci-6518/job/hsiatono-smoke-tests/32/artifact/test/functional/report/_card__add_card__Add_card_variations_should_add_card___maestro_GB.png
```

jugend committed:
fix invalid image path and jenkins url in the stacktrace

grawk commented:
This changes the structure of the object resolved from `snap`. Thus this would be a breaking change. The change looks fine to me, but we need to release it in a major version.

grawk commented:
Since this changes the resolved object from `snap`, the documentation for the object in README needs to be updated.

jugend commented:
@grawk thanks, updated the README.

Btw, I changed documentation for theJenkins fields a bit, cause the optional bracket felt like an array at one glance, moved it to comment instead.

jugend commented:
@grawk I'm thinking to change `workspaceUrl` to the existing `imageUrl` so that it is not a breaking change, what do you think? Thx.

grawk commented:
it would be a simpler merge/publish if you can make it a non-breaking feature addition. I'm in favor of that.

jugend commented:
Updated with non-breaking change. Thx!

nikulkarni commented on index.js:
```diff
@@ -133,15 +150,14 @@ module.exports = {
-        //Adding event listeners to take automatic screenshot
+        // Adding event listeners to take automatic screenshot
         if (autoCaptureOptions.indexOf('click') !== -1) {
             flow.on('scheduleTask', function (task) {
```
How come I don't see [changes](https://github.com/paypal/nemo-screenshot/blob/develop/index.js#L140) in develop in your PR?

jugend replied:
Not exactly sure, could be a bug with Github or it could be expected if the changes are not affecting the lines that have updated in develop.

nikulkarni commented on index.js:
```diff
@@ -25,16 +25,37 @@ function titleSlug(title) {
+        buildUrl = process.env.BUILD_URL,
+        jobName = process.env.JOB_NAME,
+        workspace = process.env.WORKSPACE,
+        relImagePath = screenShotPath.substr(workspace.length);
```
We assume that if `process.env.JENKINS_URL` is present, rest of the env variables are also present, however they can be very well be absent. Can we handle that scenario? Maybe just add a warning and just return

On a side note, unrelated to the PR changes, we need to add how to use these env variables in README. Don't necessarily be need to be part of this PR.

jugend replied:
@nikulkarni Yeah, we can't be absolutely sure for the variables, added the check. Thanks!

nikulkarni commented on index.js:
```diff
@@ -71,92 +114,97 @@ module.exports = {
+                    imageObj = {
+                        'imageName': null,
+                        'imagePath': null
+                    },
```
I know its not part of this PR, but this is not required IMO. We can just have `imageObj = {}`

jugend replied:
@nikulkarni You are making me work very hard for this PR :) 

Looks like my JS beautifier did the re-formatting, and brought out the the diff. Re-formatted anonymous function bracket with a space. Btw, if you guys are open with .editorconfig and some jsbeautifier rules, we could help to standardize.

nikulkarni replied:
:) yeah this is not required to be fixed as part of this PR.

grawk commented:
OK @nikulkarni are we good to merge this?

nikulkarni commented:
👍  thanks @jugend for being patient with me

jugend commented:
Thanks folks, no worries @nikulkarni :)
